backend/src/journey/service/mapper/journeyMapper.service.ts

```diff
@@ -0,0 +1,172 @@
+
+      // 3) If transportation leg: Check transportation is present.
+      && (!this.transportationMapper.isTransportationLeg(vrrLeg)
+        || this.transportationMapper.checkVrrTransportationIntegrity(vrrLeg))
+
+      // 4) If footpath leg: Check footpath information are present.
+      && (!this.transportationMapper.isFootpathLeg(vrrLeg)
+        || this.footpathMapper.checkVrrFootpathIntegrity(vrrLeg))
+
+      // 5a) Check stop sequence locations information are present.
+      // It is checked before if stop sequence is present (in DetailsMapper).
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      && this.journeyLocationMapper.checkVrrJourneyLocationsIntegrity(vrrLeg.stopSequence!)
+
+      // 5b) Check leg origin location information are present.
+      // It is checked before if origin location is present (this mapper).
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      && this.journeyLocationMapper.checkVrrLegOriginLocationIntegrity(vrrLeg.origin!)
+
+      // 5c) Check leg destination location information are present.
+      // It is checked before if destination location is present (this mapper).
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      && this.journeyLocationMapper.checkVrrLegDestinationLocationIntegrity(vrrLeg.destination!)
+    );
+  }
+
+  private checkVrrLegIntegrity(vrrLeg: VrrLeg): boolean {
+    let isValid = true;
+
+    // Ignore journeys that contain legs with missing origin and/or destination.
+    if (!vrrLeg.origin || !vrrLeg.destination) {
+
+      console.error("VRR leg is missing origin or/and destination location " +
+        "and therefore linked journey won´t be processed: " + JSON.stringify(vrrLeg));
+      isValid = false;
+    }
+
+    return isValid;
+  }
+
+  private mapVrrJourney(vrrJourney: VrrJourney): Journey {
+    const interchanges = vrrJourney.interchanges
+      ?? this.calculateVrrJourneyInterchanges(vrrJourney.legs);
+
+    const processedLegs = vrrJourney.legs
+      ?.filter(vrrLeg => !this.transportationMapper.isGesicherterAnschlussLeg(vrrLeg))
+      ?.map(vrrLeg => this.mapVrrLeg(vrrLeg));
+
+    return {
+      interchanges: interchanges ?? JOURNEY_INTERCHANGES_FALLBACK_VAL,
+      legs: processedLegs ?? JOURNEY_LEGS_FALLBACK_VAL
+    } as Journey;
+  }
+
+  private calculateVrrJourneyInterchanges(vrrLegs: (VrrLeg[] | undefined)): (number | undefined) {
+    return vrrLegs?.length && vrrLegs.length - 1 >= 0
+      ? vrrLegs.length - 1
+      : undefined;
+  }
+
+  private mapVrrLeg(vrrLeg: VrrLeg): (TransportationLeg | FootpathLeg) {
+    let leg: (TransportationLeg | FootpathLeg);
+
+    // Integrity check ensured before that origin and destination are present.
+    const baseLeg = {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      origin: this.journeyLocationMapper.mapLegOriginLocation(vrrLeg.origin!),
```

Comment:
This non-null assertion is somewhat dirty, because vrrLeg.origin doesn't even have any type that might include `undefined`. Are you sure that this assertion is really necessary? The legs are normally an array, it would be really weird if the array included an entry that is `undefined`.

Reply:
vrrLeg.origin does include undefined in its type as it is defined as optional in vrr type. 
At this line of code it is assured that this isn´t the case because of the checks that has run beforehand (in this case in line 113). My first solution was using explicit casting here (`as JourneyLocationElement`) but new linter rules require the exclamation mark instead. (That's the reason for the non-null assertion here and at some similar places.)

Reply:
Okay I see, seems like the we can't really use the info wether a property is actual optional if we get it from the VRR API types. User defined type guards and complex types or validating while mapping would solve this too, but I think this might be too much effort for now.